src/components/PopupWithForm.js

```diff
@@ -7,9 +7,17 @@ export default class PopupWithFrom extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form"); // selects modal form for specific instance of this obj
   }
 
+  setLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._popupElement.querySelector(".modal__save-button").textContent = "Saving...";
+    } else {
+      this._popupElement.querySelector(".modal__save-button").textContent = submitSave;
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`